src/components/notebook-utils.js

```diff
@@ -202,14 +196,13 @@ export class NotebookDeleter extends Component {
           this.setState({ processing: true })
           Buckets.notebook(namespace, bucketName, printName).delete().then(
             onSuccess,
-            failure => this.setState({ failure })
+            error => reportError(`Error deleting notebook: ${error}`)
```

Comment:
In this case, the spinner just sits there. Do we need a mechanism to encourage developers to cause the UI to go back to a reasonable state when an error occurs? I was hoping the try...finally pattern above would encourage this.

Reply:
I don't know if there is a 'mechanism' per-se, but we should certainly establish a habit of considering failure cases as we design new features, and try to handle them reasonably well.